plug.vim

```diff
@@ -1980,17 +1987,24 @@ function! s:update_ruby()
 EOF
 endfunction
 
-function! s:shellesc_cmd(arg)
-  let escaped = substitute(a:arg, '[&|<>()@^]', '^&', 'g')
-  let escaped = substitute(escaped, '%', '%%', 'g')
-  let escaped = substitute(escaped, '"', '\\^&', 'g')
-  let escaped = substitute(escaped, '\(\\\+\)\(\\^\)', '\1\1\2', 'g')
-  return '^"'.substitute(escaped, '\(\\\+\)$', '\1\1', '').'^"'
+function! s:shellesc_cmd(arg, script)
+  let escaped = substitute(a:arg, '%', (a:script ? '%' : '^') . '&', 'g')
+  return substitute('"'.escaped.'"', '[&|<>()@^!"]', '^&', 'g')
```

Comment:
This is copied from my vimrc. The double backslash escaping does not help on cmd.exe. Only `^` and `%` (within batchfiles)` work. I haven't had issues with this.

Reply:
`%` should be escaped last :sweat_smile: . Fixed in the latest commit.